README.md

```diff
@@ -1,47 +1,178 @@
+* [.NET](https://dotnet.microsoft.com/download)
 
-```bash
-# build and install the resource provider plugin
-$ make build install
 
-# test
+### Creating and Initializing the Repository
+
+Pulumi offers this repository as a [GitHub template repository](https://docs.github.com/en/repositories/creating-and-managing-repositories/creating-a-repository-from-a-template) for convenience.  From this repository:
+
+1. Click "Use this template".
+1. Set the following options:
+   * Owner: pulumi 
+   * Repository name: pulumi-xyz-native (replace "xyz" with the name of your provider)
+   * Description: Pulumi provider for xyz
+   * Repository type: Public
+1. Clone the generated repository.
+
+From the templated repository:
+
+1. Search-replace `xyz` with the name of your desired provider.
+
+#### Build the provider and install the plugin
+
+   ```bash
+   $ make build install
+   ```
+   
+This will:
+
+1. Create the SDK codegen binary and place it in a `./bin` folder (gitignored)
+2. Create the provider binary and place it in the `./bin` folder (gitignored)
+3. Generate the dotnet, Go, Node, and Python SDKs and place them in the `./sdk` folder
+4. Install the provider on your machine.
+
+#### Test against the example
+   
+```bash
 $ cd examples/simple
 $ yarn link @pulumi/xyz
 $ yarn install
 $ pulumi stack init test
 $ pulumi up
 ```
 
+Now that you have completed all of the above steps, you have a working provider that generates a random string for you.
+
+#### A brief repository overview
+
+You now have:
+
+1. A `provider/` folder containing the building and implementation logic
+   1. `cmd/`
+      1. `pulumi-gen-xyz/` - generates language SDKs from the schema
+      2. `pulumi-resource-xyz/` - holds the package schema, injects the package version, and starts the gRPC server
+   2. `pkg`
+      1. `provider` - holds the gRPC methods (and for now, the sample implementation logic) required by the Pulumi engine
+      2. `version` - semver package to be consumed by build processes
+2. `deployment-templates` - a set of files to help you around deployment and publication
+3. `sdk` - holds the generated code libraries created by `pulumi-gen-xyz/main.go`
+4. `examples` a folder of Pulumi programs to try locally and/or use in CI.
+5. A `Makefile` and this `README`.
+
+### Writing the schema
+
+The [JSON schema file](https://www.pulumi.com/docs/guides/pulumi-packages/schema) is used by `pulumi-gen-xyz` to create language-specific SDKs. 
+It is, therefore, a central requirement for any resource provider.
+Provider schemas can be machine-generated from an API specification, or handwritten.
+This repository provides the [xyz example schema](./provider/cmd/pulumi-resource-xyz/schema.json) to get you started.
+[The Azure Native Provider schema](https://raw.githubusercontent.com/pulumi/pulumi-azure-native/master/provider/cmd/pulumi-resource-azure-native/schema.json) provides a much larger example.
```

Comment:
I think this is a monster to create - I suggest pointing to aws-native as it's not too big to digest?